app/scripts/background.js

```diff
@@ -663,13 +699,39 @@ export function setupController(
         connectionStream: portStream,
       });
     } else {
+      // this is triggered when a new tab is opened, or origin(url) is changed
```

Comment:
well technically the url changing doesn't necessarily trigger this (single page apps like react where the url may change, but the history is changing locally only). It's really triggered after page loads.

Reply:
ah i put here for my reference as well. as it was very difficult to find where to inject and there was no comment saying what is filtered in this `if`. i would make some change in the next iteration but would love to get it a merge when finally get a green pipeline 🙏🏻